pkg/clustersync/clusterSync.go

```diff
@@ -255,3 +265,32 @@ func transformManagedCluster(managedCluster *clusterv1.ManagedCluster) model.Res
 	}
 	return resource
 }
+
+// Deletes a cluster resource and all resources from the cluster.
+func processClusterDelete(ctx context.Context, obj interface{}) {
+	klog.V(4).Info("Processing Cluster Delete.")
+	clusterName := obj.(*unstructured.Unstructured).GetName()
+	var deleteClusterNode bool
+	kind := obj.(*unstructured.Unstructured).GetKind()
+	switch kind {
+	case "ManagedCluster":
+		// When ManagedCluster (MC) is deleted, delete the resources and edges and cluster node for that cluster from db
+		// ManagedClusterInfo (namespace scoped) will be deleted when the MC (cluster scoped) is being deleted.
+		// So, we are tracking deletes of MC only to avoid duplication.
+		deleteClusterNode = true
+		klog.V(3).Infof("Received delete for %s. Deleting Cluster resource %s and all resources from the DB", kind,
+			clusterName)
+
+	case "ManagedClusterAddOn":
+		// When ManagedClusterAddOn (MCA) is deleted, search is disabled in the cluster. So, we delete the resources
+		// and edges for that cluster from db. But the cluster node is kept until MC is deleted.
+		deleteClusterNode = false
+		klog.V(3).Infof("Received delete for %s. Deleting Cluster resources and edges for cluster %s from the DB", kind,
+			clusterName)
+
+	default:
+		klog.V(4).Info("No delete cluster actions for kind: %s", kind)
```

Comment:
Maybe this log should be a warning because we don't expect this to happen.

Reply:
Added one to log ManagedClusterInfo and changed default to Warning.